libsweep/src/sweep.cc

```diff
@@ -288,111 +235,72 @@ void sweep_device_accumulate_scans(sweep_device_s device) {
       received = 1;
     }
   }
+} catch (const std::exception& e) {
+  // worker thread is dead at this point
```

Comment:
> accumulate_scans does not have an out error argument:

Given that this method is implementation ONLY and not part of the API, we could easily change its return type. But, given that `accumulate_scans` is only called from `start_scanning`, i'm unclear how we'd use the return type to propagate the error.

> that function should signal an error in the worker thread

Could you elaborate on what you meant here. The `get_scan` method is running in the main thread, not the worker thread. Worker thread just accumulate scans using `accumulate_scans`. 

> it should be possible to cancel a call to sweep_device_get_scan (and/or use a timeout)

I completely agree that we should add a timeout to `get_scan`... the specifics of which we can flesh out later. I think it might be useful to do a PR that introduces timeouts for a few different methods  (and the serial interactions) in a uniform manner.  

Also agree that this PR doesn't introduce any new behavior. If there is not an obvious way to handle the exception in this method currently, then we should push this PR 👍  and address improvements in later PRs.